### ISSUE TITLE
GridFactory crashes if no HookDispatcher was injected

### ISSUE THREAD

**Issue by jolelievre** [PrestaShop/PrestaShop#12506]
**Describe the bug**
`GridFactory` crashes when hook dispatcher was not injected.

**To Reproduce**
On the 1.7.5.x branch (1.7.5.1 target) upgrade the ps_linklist module to v3.0.1, when you try to access its configuration you get an error. This is due to a modification introduce in `GridFactory`: https://github.com/PrestaShop/PrestaShop/commit/e78714cd258c650dda1fb8a77e39a55a4bc85e55#diff-856a26096c00cb55eaf96cb866d41e7b

**Screenshots**
![capture d ecran 2019-02-08 a 11 00 28](https://user-images.githubusercontent.com/13801017/52471397-1a1ef700-2b91-11e9-852d-26f52bcd2c43.png)

**Solution**
Remove HookDispatcherAwareTrait, prefer an injection by constructor but use a `NullDispatcher` if the argument is not present:

```php
public function __construct(..., HookDistapcherInterface $dispatcher = null)
{
    ...
    $this->hookDispatcher = null !== $dispatcher ? $dispatcher : new NullDispatcher();
}
```

**Additionnal information**
PrestaShop version: 1.7.5.x
PHP version: N/A

**Comment by khouloudbelguith** [comment]
@jolelievre, same issue for me
![image](https://user-images.githubusercontent.com/16067358/52472481-b5b16700-2b93-11e9-8372-eb6b074380de.png)
Thanks!

**Comment by jolelievre** [comment]
@khouloudbelguith on branch develop or 1.7.5x ?
There is the same issue on both branch, we first fix it on 1.7.5 then we will merge the fix back to develop

**Comment by khouloudbelguith** [comment]
@jolelievre, I checked it the with 1.7.5.x & develop branch
Thanks!
